### DISCUSSION TITLE
6.4.2 프론트엔드 애플리케이션 배포 문제(react-scripts: Permission denied)

### DISCUSSION THREAD

**Discussion by heocat** [fsoftwareengineer/todo-application#119] [질문과 답변]
-현재 문제:
ㄴ배포하려는 애플리케이션 환경 상태가 server에서 ok로 변경되지 않음
![IMG_40A9E7540B2F-1](https://user-images.githubusercontent.com/96338798/170818325-8bd2f360-2976-4708-b92e-0cc19625bb14.jpeg)

-에러 로그(web.stdout.log):
`May 28 08:02:32 ip-172-31-10-222 web: sh: /var/app/current/node_modules/.bin/react-scripts: Permission denied`
(하지만 로컬에서는 권한 문제 없이 npm run start 됨)
-브라우저 상태:
502 Bad Gateway nginx/1.20.0

-시도한 방법(디스커션 참고)

1.루트 디렉토리에 .ebextensions 폴더 생성 후 01-permissions.config 파일 생성 
아래코드 삽입

**files:
  "/opt/elasticbeanstalk/hooks/appdeploy/pre/49_change_permissions.sh":
    mode: "000755"
    owner: root
    group: root
    content: |
      #!/usr/bin/env bash
      sudo chown -R ec2-user:ec2-user tmp/
      set -xe
      /opt/elasticbeanstalk/bin/healthd-track-pidfile --proxy nginx**

결과: 환경상태 변함 없음

2.ssh를 이용해서 `setfacl -R -m u:ec2-user:rwx node_modules` 로 권한 추가
참고(https://github.com/fsoftwareengineer/todo-application/discussions/35#discussion-3748280)

결과: 딱히 오류는 뜨지 않았지만 환경상태 변함 없음

p.s. 다만 참고한 페이지 댓글에서 작가님이 남겨주신 퍼미션 설정 관련 aws 페이지(https://forums.aws.amazon.com/thread.jspa?threadID=290964)를 참고하려 했으나 그 페이지가 현재시점 보관되어 확인이 불가능 했습니다.

3. ssh 접속 후 root 권한으로 

`$ sudo chmod +x node_modules/.bin/react-scripts`
`$ npm run start`

결과:eb 커맨드창에서 react-scripts: Permission denied는 뜨지 않고 run start는 되지만, 환경상태는 여전히 server에서 변하지 않음

이외의 다른 디스커션 게시글이나 스택오버플로우를 참고했으나 해결되지 않아서 글 씁니다.
알려주시면 감사하겠습니다.

**Comment by silver0-stack** [comment]
@heocat 똑같은 에러 때문에 삽질 중인데, 혹시 해당 에러 해결하셨나요?
